sdk/core/core-http/src/xhrHttpClient.ts

```diff
@@ -79,29 +79,63 @@ export class XhrHttpClient implements HttpClient {
     for (const header of request.headers.headersArray()) {
       xhr.setRequestHeader(header.name, header.value);
     }
-    xhr.responseType = request.streamResponseBody ? "blob" : "text";
+
+    const streaming =
+      (request.streamResponseStatusCodes?.size ?? 0 > 0) || request.streamResponseBody;
+    xhr.responseType = streaming ? "blob" : "text";
 
     // tslint:disable-next-line:no-null-keyword
     xhr.send(request.body === undefined ? null : request.body);
 
-    if (request.streamResponseBody) {
+    if (streaming) {
       return new Promise((resolve, reject) => {
         xhr.addEventListener("readystatechange", () => {
           // Resolve as soon as headers are loaded
           if (xhr.readyState === XMLHttpRequest.HEADERS_RECEIVED) {
-            // eslint-disable-next-line @typescript-eslint/no-shadow
-            const blobBody = new Promise<Blob>((resolve, reject) => {
+            if (request.streamResponseBody || request.streamResponseStatusCodes?.has(xhr.status)) {
+              // eslint-disable-next-line @typescript-eslint/no-shadow
+              const blobBody = new Promise<Blob>((resolve, reject) => {
+                xhr.addEventListener("load", () => {
+                  resolve(xhr.response);
+                });
+                rejectOnTerminalEvent(request, xhr, reject);
+              });
+              resolve({
+                request,
+                status: xhr.status,
+                headers: parseHeaders(xhr),
+                blobBody
+              });
+            } else {
               xhr.addEventListener("load", () => {
-                resolve(xhr.response);
+                // Response comes back in Blob when xhr.responseType === "blob"
```

Comment:
I had thought about always using `xhr.responseType === "blob"` to send request, then converting to text if needed based on the response status codes, but was back-pedaled because I don't want to make big changes to existing code paths.